src/gpt.sh

```diff
@@ -36,32 +36,33 @@ Example output:\n
 <summary>Score: 80</summary> \
 <br> \
 Improvements: \
-- some bullet points \
-<br> \
-```relevant-coding-language \
+<ul> \
+<li> some bullet points </li> \
+</ul> \r\n\r\n
+\`\`\`relevant-coding-language \
 example code here \
-``` \
+\`\`\` \
 </details>
 EOF
 )
 
 gpt::prompt_model() {
-  local -r git_diff="$1"
+  local -r git_diff="${1}"
 
-  local -r body=$(curl -sSL \
+  local -r response=$(curl -sSL \
     -H "Content-Type: application/json" \
     -H "Authorization: Bearer $OPEN_AI_API_KEY" \
     -d "$(jq -n --arg model "$GPT_MODEL" --arg prompt "$INITIAL_PROMPT" --arg git_diff "$git_diff" '{model: $model, messages: [{role: "user", content: $prompt}, {role: "user", content: $git_diff}]}')" \
     "https://api.openai.com/v1/chat/completions")
 
-  local -r error=$(echo "$body" | jq -r '.error')
+  local -r error=$(echo "$response" | jq -r '.error')
 
-  if [[ "$error" != "null" ]]; then
-    echoerr "API request failed: $error"
-    exit 1
+  if [ "$error" != "null" ]; then
+    kill -s TERM $TOP_PID
```

Comment:
Does `kill -s TERM $TOP_PID` present an issue if `$TOP_PID` env var doesn't exist?

Reply:
Oh I see you're explicitly setting it here https://github.com/Integral-Healthcare/robin-ai-reviewer/pull/35/files#diff-e4d34fc353097073f2f6ebbdc1d89cbdac9f0edf5354442c2f884839a2e8afb3R4

Reply:
Had to add this so we can exit the script when something fails in the downstream. Previously, error from within the function would still continue to execute the main script and report it as a success. So, for example, if the API key for OpenAI was incorrect, it would print the error message and still report workflow/job as successful since the main script executed successfully. Maybe that is not an issue in Github actions, but in GitLab it would falsely report the result in the UI.

Reply:
Interesting, that sounds like a gitlab issue. So long as it doesn't introduce issues with the github action flow, it seems reasonable enough

Reply:
I think GitHub workflow works the same, for example, here you have an error from bash script but it continued to show successful execution:
https://github.com/Integral-Healthcare/robin-ai-reviewer/actions/runs/6723617742/job/18274076022#step:3:131

Reply:
You're completely right. Good catch!